From 5d441cb730a3765f007491ecabf887b4d3716e3f Mon Sep 17 00:00:00 2001
From: Andrew Regenscheid <aregenscheid@gmail.com>
Date: Fri, 23 Oct 2026 01:43:47 -0400
Subject: [PATCH 2/2] Makefile

Use $(SED) for every sed call so macOS builds can pass SED=gsed.
---
 Makefile | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/Makefile b/Makefile
index ef36431..14a880b 100644
--- a/Makefile
+++ b/Makefile
@@ -16,10 +16,12 @@ BRANCH ?= $(shell git rev-parse --abbrev-ref HEAD)
 REGISTRY_AND_USERNAME := $(IMAGE_REGISTRY)/$(USERNAME)
 NAME = Talos
 
+SED ?= sed
+
 CLOUD_IMAGES_EXTRA_ARGS ?= ""
 ZSTD_COMPRESSION_LEVEL ?= 18
 
-CI_RELEASE_TAG := $(shell git log --oneline --format=%B -n 1 HEAD^2 -- 2>/dev/null | head -n 1 | sed -nE "s/^release\((.*)\):.*$$/\\1/p")
+CI_RELEASE_TAG := $(shell git log --oneline --format=%B -n 1 HEAD^2 -- 2>/dev/null | head -n 1 | $(SED) -nE "s/^release\((.*)\):.*$$/\\1/p")
 
 ARTIFACTS := _out
 
@@ -100,7 +102,7 @@ GO_VERSION ?= 1.26
 # renovate: datasource=npm depName=markdownlint-cli
 MARKDOWNLINTCLI_VERSION ?= 0.49.1
 OPERATING_SYSTEM := $(shell uname -s | tr "[:upper:]" "[:lower:]")
-ARCH := $(shell uname -m | sed 's/x86_64/amd64/' | sed 's/aarch64/arm64/')
+ARCH := $(shell uname -m | $(SED) 's/x86_64/amd64/' | $(SED) 's/aarch64/arm64/')
 TALOSCTL_DEFAULT_TARGET := talosctl-$(OPERATING_SYSTEM)
 TALOSCTL_EXECUTABLE := $(PWD)/$(ARTIFACTS)/$(TALOSCTL_DEFAULT_TARGET)-$(ARCH)
 INTEGRATION_TEST := integration-test
@@ -770,7 +772,7 @@ reproducibility-test-docker-%: $(ARTIFACTS)
 	@mkdir -p _out1/ _out2/
 	@$(MAKE) docker-$* DEST=_out1/
 	@$(MAKE) docker-$* DEST=_out2/ TARGET_ARGS="--no-cache"
-	@find _out1/ -type f | xargs -IFILE diffoscope FILE `echo FILE | sed 's/_out1/_out2/'`
+	@find _out1/ -type f | xargs -IFILE diffoscope FILE `echo FILE | $(SED) 's/_out1/_out2/'`
 	@rm -rf _out1/ _out2/
 
 reproducibility-test-local-%: $(ARTIFACTS)
@@ -778,7 +780,7 @@ reproducibility-test-local-%: $(ARTIFACTS)
 	@mkdir -p _out1/ _out2/
 	@$(MAKE) local-$* DEST=_out1/
 	@$(MAKE) local-$* DEST=_out2/ TARGET_ARGS="--no-cache"
-	@find _out1/ -type f | xargs -IFILE diffoscope FILE `echo FILE | sed 's/_out1/_out2/'`
+	@find _out1/ -type f | xargs -IFILE diffoscope FILE `echo FILE | $(SED) 's/_out1/_out2/'`
 	@rm -rf _out1/ _out2/
 
 reproducibility-test-iso: $(ARTIFACTS)
-- 
2.53.0

